service/controller/app/v1/resource/chart/desired_test.go

```diff
@@ -0,0 +1,220 @@
+				ObjectMeta: v1.ObjectMeta{
+					Name:      "giantswarm",
+					Namespace: "default",
+					Labels: map[string]string{
+						"app-operator.giantswarm.io/version": "1.0.0",
+					},
+				},
+				Spec: v1alpha1.AppCatalogSpec{
+					Title:       "Giant Swarm",
+					Description: "Catalog of Apps by Giant Swarm",
+					CatalogStorage: v1alpha1.AppCatalogSpecCatalogStorage{
+						Type: "helm",
+						URL:  "https://giantswarm.github.com/app-catalog/",
+					},
+					LogoURL: "https://s.giantswarm.io/...",
+				},
+			},
+			expectedChart: &v1alpha1.Chart{
+				TypeMeta: v1.TypeMeta{
+					Kind:       "Chart",
+					APIVersion: "application.giantswarm.io",
+				},
+				ObjectMeta: v1.ObjectMeta{
+					Name: "kubernetes-prometheus",
+					Labels: map[string]string{
+						"app":                        "prometheus",
+						"giantswarm.io/cluster":      "6iec4",
+						"giantswarm.io/organization": "giantswarm",
+						"giantswarm.io/service-type": "managed",
+					},
+					Annotations: map[string]string{
+						"giantswarm.io/managed-by":     "app-operator",
+						"giantswarm.io/version-bundle": "0.1.0",
+					},
+				},
+				Spec: v1alpha1.ChartSpec{
+					Config: v1alpha1.ChartSpecConfig{
+						ConfigMap: v1alpha1.ChartSpecConfigConfigMap{
+							Name:            "giant-swarm-config",
+							Namespace:       "giantswarm",
+							ResourceVersion: "",
+						},
+						Secret: v1alpha1.ChartSpecConfigSecret{
+							Name:            "giant-swarm-secret",
+							Namespace:       "giantswarm",
+							ResourceVersion: "",
+						},
+					},
+					Name:       "my-cool-prometheus",
+					Namespace:  "monitoring",
+					TarballURL: "https://giantswarm.github.com/app-catalog/kubernetes-prometheus-1.0.0.tgz",
+				},
+			},
+		},
+		{
+			name: "case 1: appcatalog not found",
+			obj: &v1alpha1.App{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      "my-cool-prometheus",
+					Namespace: "default",
+					Labels: map[string]string{
+						"app":                        "prometheus",
+						"giantswarm.io/cluster":      "6iec4",
+						"giantswarm.io/organization": "giantswarm",
+						"giantswarm.io/service-type": "managed",
+					},
+					Annotations: map[string]string{
+						"giantswarm.io/managed-by":     "app-operator",
+						"giantswarm.io/version-bundle": "0.1.0",
+					},
+				},
+				Spec: v1alpha1.AppSpec{
+					Catalog: "giantswarm",
+					Release: "1.0.0",
+					Config: v1alpha1.AppSpecConfig{
+						ConfigMap: v1alpha1.AppSpecConfigConfigMap{
+							Name:      "giant-swarm-config",
+							Namespace: "giantswarm",
+						},
+						Secret: v1alpha1.AppSpecConfigSecret{
+							Name:      "giant-swarm-secret",
+							Namespace: "giantswarm",
+						},
+					},
+					KubeConfig: v1alpha1.AppSpecKubeConfig{
+						Secret: v1alpha1.AppSpecKubeConfigSecret{
+							Name:      "giantswarm-12345",
+							Namespace: "12345",
+						},
+					},
+					Name:      "kubernetes-prometheus",
+					Namespace: "monitoring",
+				},
```

Comment:
I'd like to see more error cases for when the tarballURL cannot be generated.

Reply:
Should be one enough?

Reply:
I think that's fine.